get_query_generators.php

```diff
@@ -0,0 +1,336 @@
+ */
+function gen_query_questions()
+{
+    return "SELECT question_id, answer_type, content FROM questions;";
+}
+
+/**
+ * Returns sql query that returns an unordered list of sections related to a ta.
+ *
+ * @return String A sql command that returns an unorderedlist of sections.
+ */
+function gen_query_course_pairings_section($course_code, $term)
+{
+    $specify_course = "";
+    if ($course_code) {
+        $specify_course = "WHERE courses.course_code=:course_code";
+    }
+
+    $specify_term = "";
+    if ($term) {
+        $specify_term = "WHERE sections.term=:term";
+    }
+
+    $name =
+        "SELECT users.name, users.user_id FROM users WHERE user_id=:user_id";
+
+    $department = "SELECT courses.department_name, courses.course_code FROM courses $specify_course";
+
+    $sections =
+        "SELECT sections.section_id, sections.course_code, " .
+        "sections.term, sections.room, sections.section_code FROM sections $specify_term";
+
+    $ta_associations =
+        "SELECT user_associations.section_id " .
+        "FROM user_associations WHERE user_associations.user_id=:user_id";
+
+    return (
+        "SELECT t1.course_code, t1.term, t1.room, t1.section_code, t3.name, t4.department_name, t3.user_id " .
+        "FROM ($sections) t1 JOIN ($ta_associations) t2 ON t1.section_id=t2.section_id " .
+        "JOIN ($name) t3 JOIN ($department) t4 ON t4.course_code=t1.course_code;"
+    );
+}
+
+/**
+ * Returns sql query that returns an unordered list of questions.
+ *
+ * @return String A sql command that returns an unorderedlist of questions.
+ */
+function gen_query_user_role()
+{
+    return "SELECT CASE WHEN is_admin = 1 THEN 'TRUE' ELSE 'FALSE' END AS is_admin,
+    CASE WHEN is_instructor = 1 THEN 'TRUE' ELSE 'FALSE' END AS is_instructor,
+    CASE WHEN is_ta = 1 THEN 'TRUE' ELSE 'FALSE' END AS is_ta
+    FROM users WHERE user_id = :user_id;";
+}
+
+/**
+ * Returns sql query that returns an unordered list of questions.
+ *
+ * @return String A sql command that returns an unorderedlist of questions.
+ */
+function gen_query_survey_responses()
+{
+    return "SELECT question_id, GROUP_CONCAT(answer) as answers, survey_instance_id
+    FROM responses
+    WHERE survey_instance_id = :survey_id
+    GROUP BY question_id;";
+}
+
+/**
+ * Returns sql query that returns an unordered list of tas or instructors.
+ * If $ta_list is true, a list of tas is returned. If $ta_list is false,
+ * the returned list depends on column parameter in the url.
+ *
+ * For what=tas, term parameter sepcifies the term returned and course_code
+ * sepcifies the course. term and course_code parameters are optional.
+ *
+ * For what=course_pairings, column parameter is required. term and course_code
+ * are optional parameters.
+ *
+ * @param ta_list True if function call if for list of TAs
+ * @return String A sql command that returns an unorderedlist of questions.
+ */
+function gen_query_course_associations($course_code, $term, $is_ta)
```

Comment:
rename to course_pairings. Course associations makes no sense for this. Stay consistent with API naming convention.